internal/datasource/os/files_test.go

```diff
@@ -0,0 +1,235 @@
+	filePath := "/tmp/test.conf"
+	fileContent := []byte("location /test {\n    return 200 \"Test location\\n\";\n}")
+
+	tenantId, instanceId, err := createTestIds()
+	assert.NoError(t, err)
+	filesUrl := fmt.Sprintf("/instance/%s/files/", instanceId)
+
+	err = WriteFile(fileContent, filePath)
+	assert.NoError(t, err)
+	assert.FileExists(t, filePath)
+
+	metaDataTime1, err := createProtoTime("2024-01-08T14:22:21Z")
+	assert.NoError(t, err)
+
+	metaDataTime2, err := createProtoTime("2024-01-08T13:22:25Z")
+	assert.NoError(t, err)
+
+	metaDataReturn := &instances.Files{
+		Files: []*instances.File{
+			{
+				LastModified: metaDataTime1,
+				Path:         "/tmp/nginx/nginx.conf",
+				Version:      "BDEIFo9anKNvAwWm9O2LpfvNiNiGMx.c",
+			},
+			{
+				LastModified: metaDataTime2,
+				Path:         "/tmp/test.conf",
+				Version:      "Rh3phZuCRwNGANTkdst51he_0WKWy.tZ",
+			},
+		},
+	}
+
+	getFileReturn := &instances.FileDownloadResponse{
+		Encoded:     true,
+		FilePath:    "/tmp/test.conf",
+		InstanceId:  instanceId.String(),
+		FileContent: []byte("location /test {\n    return 200 \"Test location\\n\";\n}"),
+	}
+
+	fakeConfigDownloader := &client.FakeHttpConfigDownloaderInterface{}
+	fakeConfigDownloader.GetFilesMetadataReturns(metaDataReturn, nil)
+	fakeConfigDownloader.GetFileReturns(getFileReturn, nil)
+
+	fileSource := NewFileSource(&FileSourceParameters{
+		configDownloader: fakeConfigDownloader,
+	})
+
+	cacheTime1, err := createProtoTime("2024-01-08T14:22:21Z")
+	assert.NoError(t, err)
+
+	cacheTime2, err := createProtoTime("2024-01-08T12:22:21Z")
+	assert.NoError(t, err)
+
+	lastConfigApply := map[string]*instances.File{
+		"/tmp/nginx/nginx.conf": {
+			LastModified: cacheTime1,
+			Path:         "/tmp/nginx/nginx.conf",
+			Version:      "BDEIFo9anKNvAwWm9O2LpfvNiNiGMx.c",
+		},
+		"/tmp/test.conf": {
+			LastModified: cacheTime2,
+			Path:         "/tmp/test.conf",
+			Version:      "Rh3phZuCRwNGANTkdst51he_0WKWy.tZ",
+		},
+	}
+
+	currentCache, skippedFiles, err := fileSource.UpdateInstanceConfig(lastConfigApply, filesUrl, tenantId)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(skippedFiles))
+	assert.NotEqual(t, currentCache, lastConfigApply)
+
+	path := "/tmp/test.conf"
+	err = os.Remove(path)
+	assert.NoError(t, err)
+	assert.NoFileExists(t, path)
+}
+
+func createTestIds() (uuid.UUID, uuid.UUID, error) {
+	tenantId, err := uuid.Parse("7332d596-d2e6-4d1e-9e75-70f91ef9bd0e")
+	if err != nil {
+		fmt.Printf("Error creating tenantId: %v", err)
+	}
+
+	instanceId, err := uuid.Parse("aecea348-62c1-4e3d-b848-6d6cdeb1cb9c")
+	if err != nil {
+		fmt.Printf("Error creating instanceId: %v", err)
+	}
+
+	return tenantId, instanceId, err
+}
+
+func createProtoTime(timeString string) (*timestamppb.Timestamp, error) {
+	time, err := time.Parse(time.RFC3339, timeString)
+	protoTime := timestamppb.New(time)
+
+	return protoTime, err
+}
+
+func createCacheFile(t *testing.T, cachePath string) map[string]*instances.File {
```

Comment:
remove *testing.T from the function parameters. The asserts should only be done in the actual tests and not helper functions. This function should return an error of the test that calls it should assert the error.